internal/pkg/template/api/service/service.go

```diff
@@ -2,17 +2,55 @@ package service
 
 import (
 	"fmt"
+	"sync"
+	"time"
 
 	"github.com/keboola/keboola-as-code/internal/pkg/template/api/dependencies"
 	"github.com/keboola/keboola-as-code/internal/pkg/template/api/gen/templates"
+	"github.com/keboola/keboola-as-code/internal/pkg/template/repository/manager"
 )
 
 type Service struct {
-	dependencies dependencies.Container
+	dependencies        dependencies.Container
+	lock                *sync.Mutex
+	repositoriesManager *manager.Manager
 }
 
-func New(d dependencies.Container) templates.Service {
-	return &Service{dependencies: d}
+const TemplateRepositoriesPullInterval = 5 * time.Minute
+
+func New(d dependencies.Container) (*Service, error) {
+	repoManager, err := manager.New(d.Logger())
+	if err != nil {
+		return nil, err
+	}
+	s := &Service{
+		dependencies:        d,
+		lock:                &sync.Mutex{},
+		repositoriesManager: repoManager,
+	}
+	s.StartCron()
+	return s, nil
+}
+
+func (s *Service) StartCron() {
+	go func() {
+		ticker := time.NewTicker(TemplateRepositoriesPullInterval)
+		for {
+			select {
+			case <-s.dependencies.Ctx().Done():
+				return
+			case <-ticker.C:
+				s.pullTemplateRepositories()
+			}
+		}
+	}()
+}
+
+func (s *Service) pullTemplateRepositories() {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	s.repositoriesManager.Pull()
```

Comment:
Mohli by sme vypisovat, ci boli nejake zmeny, a ak ano, tak hash najnovsieho commitu?
Ak nie su zmeny, tak, ze sa zmeny nenasli?

Pomoze nam to pri debugovani, hlavne ten `hash`.

Reply:
Stačila by tahle zpráva z resetu? (Ta se ale zobrazí vždycky i když se nic nezměnilo.) Nebo zjistit hash před pullem, hash po pullu a v případě změny ho vypsat? Nebo to myslíš ještě jinak, např. doptávat se přes `git log` co se změnilo?

![CleanShot 2022-03-21 at 11 34 14](https://user-images.githubusercontent.com/215660/159244500-6defc0d2-655a-4cd2-852b-e0e58239ffff.jpg)

Reply:
No lepsie by bolo zistit ten hash a vypisat stary -> novy, ... ale nemusis to robit teraz, ak mas este ine tasky co chces dokoncit, tak staci ze zalozis taks a doplni sa to v dalsom sprinte, a toto mergneme bez toho.

Reply:
`git rev-parse HEAD`

Reply:
Ok, uzavřel bych teď tu hlavní funkcionalitu a tohle přidal jako vylepšení. Klidně začnu hned po mergi. A nebo jít nejdřív na ten error handling. Nevím jak moc toho máš nachystanýho a chceš to dotáhnout (říkal jsi myslím že už jsi k tomu chystal to předávání exception id ze Storage), tak si spíš asi řekni.

Reply:
Ten error handling by som zobral ja, mozes mi ho prehodit, mal by som to mat za 1-2h hotove (ked doriesim veci in progress).

Ak by si uz nemal vobec ziaden task, tak budeme potrebovat riesit e2e testy API.